pkg/actors/actors.go

```diff
@@ -762,90 +753,79 @@ func (a *actorsRuntime) startReminder(reminder *Reminder, stopChannel chan bool)
+					reminder.Name, reminder.ActorType, reminder.ActorID, err)
+			} else {
+				repetitions.decrementAttempt()
+				if err = a.updateReminderTrack(actorKey, reminder.Name, repetitions.getValue()); err != nil {
+					log.Errorf("error updating reminder track: %v", err)
 				}
-			}(t, reminder.ActorType, reminder.ActorID, reminder.Name, reminder.DueTime, reminder.Period, repetitions, reminder.Data)
-		} else {
-			err := a.DeleteReminder(context.TODO(), &DeleteReminderRequest{
-				Name:      reminder.Name,
-				ActorID:   reminder.ActorID,
-				ActorType: reminder.ActorType,
-			})
-			if err != nil {
-				log.Errorf("error deleting reminder: %s", err)
+			}
+			// check if the reminder is repetitive
+			if period == 0 {
+				break
+			}
+			select {
+			case <-time.After(period):
+			case <-stop:
+				return
 			}
 		}
+		err := a.DeleteReminder(context.TODO(), &DeleteReminderRequest{
+			Name:      reminder.Name,
+			ActorID:   reminder.ActorID,
+			ActorType: reminder.ActorType,
+		})
+		if err != nil {
+			log.Errorf("error deleting reminder: %s", err)
+		}
 	}(reminder, period, repetitions, stopChannel)
 
 	return nil
 }
 
-func (a *actorsRuntime) executeReminder(actorType, actorID, dueTime, period, reminder string, repetition *actorRepetition, data interface{}) error {
+func (a *actorsRuntime) executeReminder(reminder *Reminder) error {
 	r := ReminderResponse{
-		DueTime: dueTime,
-		Period:  period,
-		Data:    data,
+		DueTime: reminder.DueTime,
+		Period:  reminder.Period,
+		Data:    reminder.Data,
 	}
 	b, err := json.Marshal(&r)
 	if err != nil {
 		return err
 	}
 
-	log.Debugf("executing reminder %s for actor type %s with id %s", reminder, actorType, actorID)
-	req := invokev1.NewInvokeMethodRequest(fmt.Sprintf("remind/%s", reminder))
-	req.WithActor(actorType, actorID)
+	log.Debugf("executing reminder %s for actor type %s with id %s", reminder.Name, reminder.ActorType, reminder.ActorID)
+	req := invokev1.NewInvokeMethodRequest(fmt.Sprintf("remind/%s", reminder.Name))
+	req.WithActor(reminder.ActorType, reminder.ActorID)
 	req.WithRawData(b, invokev1.JSONContentType)
 
 	_, err = a.callLocalActor(context.Background(), req)
-	if err == nil {
-		key := constructCompositeKey(actorType, actorID)
-		repetition.decrementAttempt()
-		err = a.updateReminderTrack(key, reminder, repetition.getValue())
-	} else {
-		log.Errorf("error execution of reminder %s for actor type %s with id %s: %s", reminder, actorType, actorID, err)
-	}
 	return err
 }
 
 func (a *actorsRuntime) reminderRequiresUpdate(req *CreateReminderRequest, reminder *Reminder) bool {
 	if reminder.ActorID == req.ActorID && reminder.ActorType == req.ActorType && reminder.Name == req.Name &&
-		(reminder.Data != req.Data || reminder.DueTime != req.DueTime || reminder.Period != req.Period) {
+		(reminder.Data != req.Data || reminder.DueTime != req.DueTime || reminder.Period != req.Period || len(req.TTL) != 0) {
```

Comment:
Also check if existing reminder has TTL. It would be the scenario where the reminder has TTL but the req does not, so an update would still be required.

Reply:
good catch, thanks!